Modify: Warp.Warp version v0.2025.02.24.20.50.stable_04

SpecterShell opened the pull request:
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/233165)

SpecterShell committed:
Update: Warp.Warp version v0.2025.02.24.20.50.stable_04

SpecterShell committed:
Add Dependencies

SpecterShell commented:
@nichoj10 appearing that you are part of the developer team, I would bring this PR to your attention.

Just out of curiosity: Why the update was disabled? For non-portable packages WinGet does not record the version installed through WinGet, but instead directly read the version string from the registry written by the installer. So if the app updates itself (and correctly update the version in the registry), WinGet won't update it again.

SpecterShell commented:
@stephengillie https://github.com/microsoft/winget-pkgs/issues/152555

```
2025-02-26 14:29:51.269 [CLI ] Installer [X64,burn,Machine,] not applicable: Installer scope does not match required scope: Machine != User
2025-02-26 14:29:51.271 [CLI ] No suitable installer found for manifest Microsoft.VCRedist.2015+.x64 with version 14.42.34438.0
2025-02-26 14:29:51.272 [CLI ] Terminating context: 0x8a150104 at D:\a\_work\1\s\external\pkg\src\AppInstallerCLICore\Workflows\DependenciesFlow.cpp:11a
```

stephengillie commented:
Automatic Validation ended with:
>  Installation failed with exit code -1978334972 2025-02-26 14:29:51.271 [CLI ] No suitable installer found for manifest Microsoft.VCRedist.2015+.x64 with version 14.42.34438.0

Automated error analysis suggests -1978334972 may mean This package has a dependency missing from your system.

(Automated response - build 986.)

stephengillie commented:
This PR doesn't appear to modify the installer files.

nichoj10 commented:
> @nichoj10 appearing that you are part of the developer team, I would bring this PR to your attention.
> 
> Just out of curiosity: Why the update was disabled? For non-portable packages WinGet does not record the version installed through WinGet, but instead directly read the version string from the registry written by the installer. So if the app updates itself (and correctly update the version in the registry), WinGet won't update it again.

I'm not sure. What's the change you're proposing? @SpecterShell

SpecterShell commented:
> > @nichoj10 appearing that you are part of the developer team, I would bring this PR to your attention.
> > Just out of curiosity: Why the update was disabled? For non-portable packages WinGet does not record the version installed through WinGet, but instead directly read the version string from the registry written by the installer. So if the app updates itself (and correctly update the version in the registry), WinGet won't update it again.
> 
> I'm not sure. What's the change you're proposing? @SpecterShell

Added `ProductCode` to help WinGet match ARP entries more accurately.

Added `Scope` so WinGet can automatically choose the correct scope (`user`/`machine`) and apply corresponding switches (`/CURRENTUSER`/`/ALLUSERS`).

![图片](https://github.com/user-attachments/assets/ed6e0af6-6739-490c-85c2-ed441faabfd8)

Added `Dependencies` (VCRedist) to resolve the following issue:

![image](https://private-user-images.githubusercontent.com/114952565/417214255-307a16dd-0235-44f5-94ff-b219efb82e7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE0MTMwNjAsIm5iZiI6MTc0MTQxMjc2MCwicGF0aCI6Ii8xMTQ5NTI1NjUvNDE3MjE0MjU1LTMwN2ExNmRkLTAyMzUtNDRmNS05NGZmLWIyMTllZmI4MmU3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwOFQwNTQ2MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YTYyNDVhMGI2NTQ0MDgzMjExMmMwMWQ3ZmI5ZWU2MDgyMDlmYTcwNWI5ZDM0YzI1NzMzOGFjNDllZmM3MTYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._te4YJi8nr216k-4d4B06d-CB5Hi7TCH-W0AQUScu3s)

This may raise some errors in the validation pipeline due to #152555. You can tag `https://github.com/stephengillie` on that.

nichoj10 commented:
> > > @nichoj10 appearing that you are part of the developer team, I would bring this PR to your attention.
> > > Just out of curiosity: Why the update was disabled? For non-portable packages WinGet does not record the version installed through WinGet, but instead directly read the version string from the registry written by the installer. So if the app updates itself (and correctly update the version in the registry), WinGet won't update it again.
> > 
> > 
> > I'm not sure. What's the change you're proposing? @SpecterShell
> 
> Added `ProductCode` to help WinGet match ARP entries more accurately.
> 
> Added `Scope` so WinGet can automatically choose the correct scope (`user`/`machine`) and apply corresponding switches (`/CURRENTUSER`/`/ALLUSERS`). The other switches can be omitted as WinGet will automatically apply them when the `InstallerType` is set to `inno`.
> 
> ![图片](https://private-user-images.githubusercontent.com/56779163/420564798-ed6e0af6-6739-490c-85c2-ed441faabfd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxOTg0NTcsIm5iZiI6MTc0NjE5ODE1NywicGF0aCI6Ii81Njc3OTE2My80MjA1NjQ3OTgtZWQ2ZTBhZjYtNjczOS00OTBjLTg1YzItZWQ0NDFmYWFiZmQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAyVDE1MDIzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3MTNmNDkxMGFkNjUxZjEyYWQ0NWNkNDM1YTBjODAxNzgzM2ZiZTlkYTg1YWYwYzcyNDc0M2I4NmEwMzYzMzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qig5Exi9zuz_8vqdVJDqLa3J_v9MZ9Wh9bn11UUVNUQ)
> 
> Added `Dependencies` (VCRedist) to resolve the following issue:
> 
> ![image](https://private-user-images.githubusercontent.com/114952565/417214255-307a16dd-0235-44f5-94ff-b219efb82e7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE0MTMwNjAsIm5iZiI6MTc0MTQxMjc2MCwicGF0aCI6Ii8xMTQ5NTI1NjUvNDE3MjE0MjU1LTMwN2ExNmRkLTAyMzUtNDRmNS05NGZmLWIyMTllZmI4MmU3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwOFQwNTQ2MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YTYyNDVhMGI2NTQ0MDgzMjExMmMwMWQ3ZmI5ZWU2MDgyMDlmYTcwNWI5ZDM0YzI1NzMzOGFjNDllZmM3MTYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._te4YJi8nr216k-4d4B06d-CB5Hi7TCH-W0AQUScu3s)
> 
> This may raise some errors in the validation pipeline due to #152555. You can tag https://github.com/stephengillie on that.

I meant to ask specifically about the update being disabled. When you try to update Warp via winget, you get this:

![image (5)](https://github.com/user-attachments/assets/d1848f60-afed-4fbc-8523-b7768e1cbe9c)

I think it's because we have `Upgrade: deny` and `RequiresExplicitUpgrade:true` in our manifest. Is that right? @SpecterShell

SpecterShell commented:
> I meant to ask specifically about the update being disabled. When you try to update Warp via winget, you get this:
> 
> ![image (5)](https://private-user-images.githubusercontent.com/56044916/439961158-d1848f60-afed-4fbc-8523-b7768e1cbe9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxOTg3NTMsIm5iZiI6MTc0NjE5ODQ1MywicGF0aCI6Ii81NjA0NDkxNi80Mzk5NjExNTgtZDE4NDhmNjAtYWZlZC00ZmJjLTg1MjMtYjc3NjhlMWNiZTljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAyVDE1MDczM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4OGM2NDRhZjlkYzJlNjg3MGNlYTM3NzJlYTEwZmJlMmFhM2RlN2UyZGM4NGUwMjMzMzU3MWQxNjZjNjE1MzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eH-Oiw9GQzCx_LGMyN1cVI2qzjr4WWMiT2yiKpB3HFU)
> 
> I think it's because we have `Upgrade: deny` and `RequiresExplicitUpgrade:true` in our manifest. Is that right? @SpecterShell

It is caused by `UpgradeBehavior: deny`. `RequiresExplicitUpgrade: true` only excludes the package from being updated by `winget upgrade --all`, but still allows updating by running `winget upgrade Warp.Warp` explicitly.

nichoj10 commented:
> > I meant to ask specifically about the update being disabled. When you try to update Warp via winget, you get this:
> > ![image (5)](https://private-user-images.githubusercontent.com/56044916/439961158-d1848f60-afed-4fbc-8523-b7768e1cbe9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxOTg3NTMsIm5iZiI6MTc0NjE5ODQ1MywicGF0aCI6Ii81NjA0NDkxNi80Mzk5NjExNTgtZDE4NDhmNjAtYWZlZC00ZmJjLTg1MjMtYjc3NjhlMWNiZTljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAyVDE1MDczM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4OGM2NDRhZjlkYzJlNjg3MGNlYTM3NzJlYTEwZmJlMmFhM2RlN2UyZGM4NGUwMjMzMzU3MWQxNjZjNjE1MzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eH-Oiw9GQzCx_LGMyN1cVI2qzjr4WWMiT2yiKpB3HFU)
> > I think it's because we have `Upgrade: deny` and `RequiresExplicitUpgrade:true` in our manifest. Is that right? @SpecterShell
> 
> It is caused by `UpgradeBehavior: deny`. `RequiresExplicitUpgrade: true` only excludes the package from being updated by `winget upgrade --all`, but still allows updating by running `winget upgrade Warp.Warp` explicitly.

Hey @SpecterShell! I opened a PR to remove those lines. I hope this PR goes through and then we can update from `winget upgrade --all` and `winget upgrade Warp.Warp`: https://github.com/microsoft/winget-pkgs/pull/255784

nichoj10 commented:
The change landed, but it's not functional. My guess is that `winget` won't compare our version strings (e.g., v0.2025.05.07.08.12.stable_02) to know which is the latest.

My Warp app is not on the latest version and yet:
```
winget upgrade Warp.Warp
No available upgrade found.
No newer package versions are available from the configured sources.
```
fyi @SpecterShell

SpecterShell commented:
Works normally in my VM. 🤔

![](https://github.com/user-attachments/assets/f3103ec5-583c-4d7a-894a-7faee38c0968)

nichoj10 commented:
@SpecterShell oops I was just premature on testing before the new version was available. It's working great now 😄
